src/api_client/api_client.js

```diff
@@ -69,29 +59,36 @@ export default class APIClient {
 	}
 
 	static createMainnetAPIClient(options) {
-		return new APIClient(MAINNET_NODES, MAINNET_NETHASH, options);
+		return new APIClient(
+			MAINNET_NODES,
+			Object.assign({}, options, { nethash: MAINNET_NETHASH }),
+		);
 	}
 
 	static createTestnetAPIClient(options) {
-		return new APIClient(TESTNET_NODES, TESTNET_NETHASH, options);
+		return new APIClient(
+			TESTNET_NODES,
+			Object.assign({}, options, { nethash: TESTNET_NETHASH }),
+		);
 	}
 
 	static createBetanetAPIClient(options) {
-		return new APIClient(BETANET_NODES, BETANET_NETHASH, options);
+		return new APIClient(
+			BETANET_NODES,
+			Object.assign({}, options, { nethash: BETANET_NETHASH }),
+		);
 	}
 
-	initialize(nodes, nethash, providedOptions = {}) {
+	initialize(nodes, providedOptions = {}) {
 		if (!Array.isArray(nodes) || nodes.length <= 0) {
 			throw new Error('APIClient requires nodes for initialization.');
 		}
 
-		if (typeof nethash !== 'string' || nethash === '') {
```

Comment:
We might still want a minimal check like this even if it's provided in options. E.g. somebody passes an array...